validations/src/validations.rs

```diff
@@ -244,23 +247,50 @@ pub fn validate_rad_request(rad_request: &RADRequest) -> Result<(), failure::Err
     if retrieval_paths.is_empty() {
         return Err(DataRequestError::NoRetrievalSources.into());
     }
+    let mut is_rng = false;
     for path in retrieval_paths {
+        if path.kind == RADType::Rng {
+            if path.url.is_empty() || path.script.is_empty() {
+                is_rng = true;
+                break;
+            } else {
+                return Err(DataRequestError::InvalidRngRequest.into());
+            }
```

Comment:
I don't really follow the logic here hehehe.

- I can see why `url` must be empty (won't be used).
- I have doubts on whether `script` should be allowed or not.
- By reading this code, I understand that as long as `url` is empty, `script` can be non-empty, but will it be used anyway?

Reply:
Some general but related comments:

- `RADTypes` only exist at the data source / retrieval level. We should avoid handling a data request differently only because one of the data sources is of a special kind.
- There won't exist such thing as RNG requests, only requests with RNG sources.
- RNG and HttpGet sources should coexist in the same data request as long as they have the same output type.

Reply:
E.g. in the future we may have a data source type that performs general computation by running WASM bytecode in a sandbox.

And you may want to use that side by side with a classic HttpGet data source that triggers a similar computation on some cloud, or fetches similar outputs from some API.

My point is that you should perfectly be able to do so, and this part of the code should only care about routing the data source definitions to their "resolvers" (the resolver of HttpGet is our wrapper of the Surf library, the resolver of Rng is the RNG and the resolver of the hypothetical Wasm type would be a WASM virtual machine).

Reply:
Agree that `is_rng` is a hack and we should find a way to avoid it.

Reply:
It should be and AND